CDCgov/prime-reportstream
# Jessica/8095 implement condition filter

This PR implements the condition filter. 

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. In organizations.yml, update the Ignore FULL_ELR receiver to have the following condition filter, replacing 600-7 with whatever loinc code shows up in your Postman test: 
               conditionFilter: ["%obsPerformedCodes.intersect('600-7').exists()"]
2. For your local env to pick up the changes that were made in this PR to the fhirpath_filter_shorthand.csv file you need to:
3. ./gradlew resetDB
4. ./gradlew run
5. ./gradlew reloadTables
6. To get your local env to pick up the changes that were made by you in your organizations.yml file you now need to run: 
            ./gradlew reloadSettings
8. Make sure when you run Send HCA HL7 Full ELR it submits and then when you run Submissions by report ID with the report ID from the response body in the first call that after at least a minute you see that sentReports is populated.
9. Change the LOINC code to a code that does not exist in the body of your call and follow the above steps. Make sure that you see filteredReportItems in the Submissions by Report ID. 
9. For good measure, add one that is not found followed by one that is and check that it still is sent
10. Send the Covid Postman test and make sure that it is not affected. 

## Changes
- *Include a comprehensive list of changes in this PR*
Implemented the condition filter
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue 8095

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints? No
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints? NO
- Does this change introduce new dependencies that need vetting? No
- Does this change require changes to our infrastructure? No
- Does logging contain sensitive data? No
- Does this PR include or remove any sensitive information itself? No

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
